kubespawner/spawner.py

```diff
@@ -479,6 +479,35 @@ def _hub_connect_port_default(self):
         """
     )
 
+    extra_objects_hook = Callable(
+        None,
+        allow_none=True,
+        config=True,
+        help="""
+        Callable to create extra objects before launching the Pod.
+
+        Expects a callable that takes one parameters:
+
+          1. The spawner object that is doing the spawning
+
+        It should return a iterable of dictionaries with the following keys:
+
+          1. object - The Kubernetes object to create, instantiated from the kubernetes client library
+          2. kind - The `kind` of `object`. For example, 'Pod', 'Deployment', 'Namespace', etc
+          3. api_version: The `apiVersion` of object to be created.
+          4. namespaced (bool): True (default) if the object is namespaced (like pods), false otherwise
+          5. ifexists: Action to take if object already exists. Possible options are: ignore, patch (default) & error.
```

Comment:
do we have a use-case for patch? I feel like it might not do what people expect. Maybe ignore/replace are sufficient?

Reply:
How would this work on the long run without patch? If I changed which service account or namespace annotation (for example) gets created, then changing every object created until then is a tedious manual operation...

But in some other cases (like PVCs) patch is probably not what you want.